$OpenBSD: patch-lib_Mail_SpamAssassin_DnsResolver_pm,v 1.1 2014/09/30 13:06:24 okan Exp $

# r1603518 - fix DNS Resolving to fix the issues with Net::DNS >= 0.76

--- lib/Mail/SpamAssassin/DnsResolver.pm.orig	Fri Feb  7 03:36:28 2014
+++ lib/Mail/SpamAssassin/DnsResolver.pm	Mon Sep 29 15:13:35 2014
@@ -204,8 +204,10 @@ sub configured_nameservers {
     @ns_addr_port = @{$self->{conf}->{dns_servers}};
     dbg("dns: servers set by config to: %s", join(', ',@ns_addr_port));
   } elsif ($res) {  # default as provided by Net::DNS, e.g. /etc/resolv.conf
-    @ns_addr_port = map(untaint_var("[$_]:" . $res->{port}),
-                        @{$res->{nameservers}});
+    my @ns = $res->UNIVERSAL::can('nameservers') ? $res->nameservers
+						 : @{$res->{nameservers}};
+    my $port = $res->UNIVERSAL::can('port') ? $res->port : $res->{port};
+    @ns_addr_port = map(untaint_var("[$_]:" . $port), @ns);
     dbg("dns: servers obtained from Net::DNS : %s", join(', ',@ns_addr_port));
   }
   return @ns_addr_port;
